File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/cot/Java/Python/s456888268_commented.py
def solve():
    tmp = ns()
    s = list(tmp)
    n = len(s)
    ans = 0
    cnt = 0
    for i in range(n):
        if s[i] == 'S':
            cnt += 1
        else:
            if cnt == 0:
                continue
            ans += 1
            cnt -= 1
    print(n - 2 * ans)

# Input methods
def ns():
    return input().strip()

# Main execution
if __name__ == "__main__":
    solve()
# End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/mad/Java/Go/s456888268_commented.go
package main

import (
	"fmt"
)

func solve(s string) {
	ans := 0
	cnt := 0
	for _, c := range s {
		if c == 'S' {
			cnt++
		} else {
			if cnt == 0 {
				continue
			}
			ans++
			cnt--
		}
	}
	fmt.Println(len(s) - 2*ans)
}

func main() {
	var s string
	fmt.Scanln(&s)
	solve(s)
}
// End of Code
